File: libs/monitoring/features/set-scan-policy/set-scan-policy.command.ts
import type { TenantId, WorkspaceId } from '@social-monitor/shared-kernel';

export type SetScanPolicyCommand = {
  readonly tenantId: TenantId;
  readonly workspaceId: WorkspaceId;
  readonly sourceBindingId: string;
  readonly intervalSeconds: number;
  readonly freshnessSeconds: number;
  readonly retryBudget: number;
  readonly idempotencyKey: string;
  readonly correlationId: string;
};

File: apps/frontend/features/sources/lib/src/application/commands/set_scan_policy_command.dart
import 'package:social_monitor_shared_kernel/social_monitor_shared_kernel.dart';

import '../../domain/value_objects/source_binding_id.dart';

final class SetScanPolicyCommand {
  const SetScanPolicyCommand({
    required this.scope,
    required this.sourceBindingId,
    required this.intervalSeconds,
    required this.freshnessSeconds,
    required this.retryBudget,
    required this.idempotencyKey,
  });

  final WorkspaceScope scope;
  final SourceBindingId sourceBindingId;
  final int intervalSeconds;
  final int freshnessSeconds;
  final int retryBudget;
  final String idempotencyKey;
}
